chef/cookbooks/ceph/recipes/conf.rb

```diff
@@ -89,11 +89,58 @@
     is_rgw: is_rgw,
     rgw_port: rgw_port,
     rgw_pemfile: rgw_pemfile,
-    keystone_settings: keystone_settings
+    keystone_settings: node["platform_family"] == "suse" ? {} : keystone_settings
   )
   mode "0644"
 end
 
+if is_rgw && node["platform_family"] == "suse"
+  # We create a separate ceph.conf file for use only by the radosgw daemon,
+  # and an override systemd unit file which makes radosgw use this config file,
+  # so we can limit its permissions as it may include the keystone admin token.
+  #
+  # It's not sufficient to just set the main ceph.conf to 0640, because other
+  # services (glance/cinder/nova) may need to read ceph.conf.
+  #
+  # Note that ceph.conf still includes a radosgw section, because that's necssary
+  # for the ceph-radosgw-prestart.sh script to read.  ceph-radosgw.conf is thus
+  # almost identical to ceph.conf, but it also includes the keystone settings
+  # if applicable.
+  #
+  # TODO: might want the SOC supportconfig plugin to pick up this file too
+  template "/etc/ceph/ceph-radosgw.conf" do
+    source "ceph.conf.erb"
+    variables(
+      mon_initial: mon_init,
+      mon_addresses: mon_addr,
+      pool_size: rep_num,
+      pool_pg_num: pg_num,
+      osd_nodes_count: osd_nodes.length,
+      public_network: node["ceph"]["config"]["public-network"],
+      cluster_network: node["ceph"]["config"]["cluster-network"],
+      is_rgw: is_rgw,
+      rgw_port: rgw_port,
+      rgw_pemfile: rgw_pemfile,
+      keystone_settings: keystone_settings
+    )
+    owner "root"
+    group node["ceph"]["radosgw"]["group"]
+    mode "0640"
+  end
+
+  # When generating the override systemd unit file, we have to disable the
+  # radosgw service so that it gets re-enabled later, thus picking up the
+  # override unit file (if we didn't do this, a host that was already
+  # running radosgw and was then upgraded wouldn't pick up the new unit file).
+  bash "generating override ceph-radosgw systemd unit file" do
+    code <<-EOH
+      sed -e 's/\\(ExecStart=.*\\)\\(\\.conf.*\\)/\\1-radosgw\\2/' \
+        /usr/lib/systemd/system/ceph-radosgw@.service > /etc/systemd/system/ceph-radosgw@.service
+      systemctl disable #{node["ceph"]["radosgw"]["service_name"]}
```

Comment:
Oh, wait, I should probably *also* do `systemctl daemon-reload`, because the `sed` can potentially change the override unit file if the stock unit file has changed in the meantime... Shall fix.